server/service/global_policies.go

```diff
@@ -609,9 +609,32 @@ func (e AutofillError) Internal() string {
 }
 
 func (svc *Service) AutofillPolicySql(ctx context.Context, sql string) (description string, resolution string, err error) {
+	vc, ok := viewer.FromContext(ctx)
+	if !ok {
+		svc.authz.SkipAuthorization(ctx)
+		return "", "", fleet.ErrNoContext
+	}
+
 	// We expect that only users with policy write permissions will autofill policies.
-	if err = svc.authz.Authorize(ctx, &fleet.Policy{}, fleet.ActionWrite); err != nil {
-		return "", "", err
+	if vc.User.GlobalRole != nil || len(vc.User.Teams) == 0 {
+		if err = svc.authz.Authorize(ctx, &fleet.Policy{}, fleet.ActionWrite); err != nil {
+			return "", "", err
+		}
+	} else {
+		// Check if this user has team policy write permissions.
+		teamID := vc.User.Teams[0].Team.ID
+		for _, teamUser := range vc.User.Teams {
+			if teamUser.Role == fleet.RoleAdmin || teamUser.Role == fleet.RoleMaintainer || teamUser.Role == fleet.RoleGitOps {
+				teamID = teamUser.Team.ID
+				break
+			}
+		}
+		err = svc.authz.Authorize(
+			ctx, &fleet.Policy{PolicyData: fleet.PolicyData{TeamID: &teamID}}, fleet.ActionWrite,
+		)
+		if err != nil {
+			return "", "", err
+		}
```

Comment:
Another approach (more future proof in case we add a new role in the future that can also write policies) is to run `svc.authz.Authorize` on the loop until it succeeds on one (or fails on all).

Am ok with leaving as is given we are close to releasing.